packages/compass-collection/src/components/collection-tab.tsx

```diff
@@ -106,7 +138,32 @@ const CollectionTabWithMetadata: React.FunctionComponent<
           <TabNavBar
             data-testid="collection-tabs"
             aria-label="Collection Tabs"
-            tabs={tabs.map((tab) => {
+            tabNames={tabs.map((tab) => tab.name)}
+            tabLabels={tabs.map((tab) => {
+              // We don't show stats, when the collection is a timeseries or a view
+              // or when the view is being edited
+              const hideStats =
+                collectionMetadata.isTimeSeries ||
+                collectionMetadata.sourceName ||
+                editViewName;
+              if (!hideStats && tab.name === 'Documents') {
```

Comment:
Super nit, if you first check for hideStats, you don't need to do it here and in the check below, seems to be that it reads a bit easier that way:

```tsx
if (alwaysHideStats) {
  return tab.name;
}
if (tab.name === 'Documents') {
  ...
```